app/src/main/java/ayds/lisboa/songinfo/moredetails/fulllogic/OtherInfoWindow.kt

```diff
@@ -0,0 +1,186 @@
+package ayds.lisboa.songinfo.moredetails.fulllogic
+
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import android.text.Html
+import android.view.View
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import ayds.lisboa.songinfo.R
+import com.google.gson.Gson
+import com.google.gson.JsonElement
+import com.google.gson.JsonObject
+import com.squareup.picasso.Picasso
+import retrofit2.Retrofit
+import retrofit2.converter.scalars.ScalarsConverterFactory
+import java.util.*
+
+
+private const val IMAGE_URL = "https://upload.wikimedia.org/wikipedia/commons/thumb/d/d4/Lastfm_logo.svg/320px-Lastfm_logo.svg.png"
+private const val RETROFIT_URL = "https://ws.audioscrobbler.com/2.0/"
+private const val ARTIST = "artist"
+private const val BIO = "bio"
+private const val CONTENT = "content"
+private const val URL = "url"
+private const val HTML_OPENING_TAG = "<html><div width=400><font face=\"arial\">"
+private const val HTML_CLOSING_TAG = "</font></div></html>"
+
+class OtherInfoWindow : AppCompatActivity() {
+    private lateinit var artistTextView: TextView
+    private lateinit var dataBase: DataBase
+    private lateinit var retrofit: Retrofit
+    private lateinit var lastFMAPI: LastFMAPI
+    private lateinit var artistName: String
+    private lateinit var imageView: ImageView
+    private lateinit var openUrlButton: View
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_other_info)
+
+        initProperties()
+        initAPI()
+        initDataBase()
+        initIntentData()
+
+        updateArtistInfoAsync()
+    }
+
+    private fun initProperties(){
+        artistTextView = findViewById(R.id.artistInfoTextView)
+        imageView = findViewById<View>(R.id.imageView) as ImageView
+        openUrlButton = findViewById(R.id.openUrlButton)
+    }
+
+    private fun initIntentData(){
+        artistName = intent.getStringExtra("artistName") ?: ""
+    }
+
+    private fun initAPI() {
+        retrofit = createRetrofit()
```

Comment:
la variable retrofit no se utiliza más que acá, se puede eliminar de las propiedades
```suggestion
        val retrofit = createRetrofit()
```